x-pack/plugins/dataset_quality/server/routes/data_streams/get_malformed_docs.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+import { rangeQuery, termQuery } from '@kbn/observability-plugin/server';
+import { MalformedDocs } from '../../../common/api_types';
+import {
+  DATA_STREAM_DATASET,
+  DATA_STREAM_NAMESPACE,
+  DATA_STREAM_TYPE,
+  _IGNORED,
+} from '../../../common/es_fields';
+import { DataStreamTypes } from '../../types/data_stream';
+import { createDatasetQualityESClient, wildcardQuery } from '../../utils';
+
+export async function getMalformedDocsPaginated(options: {
+  esClient: ElasticsearchClient;
+  type?: DataStreamTypes;
+  start: number;
+  end: number;
+  datasetQuery?: string;
+  after?: {
+    dataset: string;
+    namespace: string;
+  };
+  prevResults?: MalformedDocs[];
+}): Promise<MalformedDocs[]> {
+  const { esClient, type = 'logs', datasetQuery, start, end, after, prevResults = [] } = options;
+
+  const datasetQualityESClient = createDatasetQualityESClient(esClient);
+
+  const response = await datasetQualityESClient.search({
```

Comment:
for future: I think it would be a great idea to add some latency telemetry for this query.